### PR TITLE
Catch PDOExceptions and check error codes

### PR DESCRIPTION
**Pull request by dragosprotung**
PDOExceptions are now caught and pdo error codes are checked.
This follows the discussion from #77

### CODE REVIEW

File: tests/TestUtil.php [hunk 1]
```diff
@@ -45,7 +45,7 @@ public static function getConnection(): PDO
             $dsn .= 'dbname=' . $connectionParams['dbname'] . $separator;
             $dsn .= self::getCharsetValue($connectionParams['charset'], $connectionParams['driver']) . $separator;
             $dsn = rtrim($dsn);
-            self::$connection = new PDO($dsn, $connectionParams['user'], $connectionParams['password']);
+            self::$connection = new PDO($dsn, $connectionParams['user'], $connectionParams['password'], [PDO::ATTR_ERRMODE => PDO::ERRMODE_EXCEPTION]);
```
Comment (dragosprotung) [inline tests/TestUtil.php]: @prolic should the error mode be an env variable ? should we have by default silent mode but add another entry in the matrix for exception mode ?

Comment (prolic) [inline tests/TestUtil.php]: that would be nice!

File: src/Projection/PdoEventStoreProjector.php [hunk 2]
```diff
@@ -501,7 +502,11 @@ private function fetchRemoteStatus(): ProjectionStatus
 EOT;
 
         $statement = $this->connection->prepare($sql);
-        $statement->execute([$this->name]);
+        try {
+            $statement->execute([$this->name]);
+        } catch (PDOException $exception) {
+            // ignore
```
Comment (prolic) [inline src/Projection/PdoEventStoreProjector.php]: consistent: // ignore and check error code

File: src/Projection/PdoEventStoreReadModelProjector.php [hunk 3]
```diff
@@ -462,7 +462,11 @@ private function fetchRemoteStatus(): ProjectionStatus
 EOT;
 
         $statement = $this->connection->prepare($sql);
-        $statement->execute([$this->name]);
+        try {
+            $statement->execute([$this->name]);
+        } catch (PDOException $exception) {
+            // ignore
```
Comment (prolic) [inline src/Projection/PdoEventStoreReadModelProjector.php]: // ignore and check error code

**Comment by prolic** [comment]
Missing try-catches:

MySqlEventStore.php:94
MySqlEventStore.php:116
MySqlEventStore.php:135
MySqlEventStore.php:397
MySqlEventStore.php:688
PdoStreamIterator.php:152
PdoStreamIterator.php:193
PostgresEventStore.php:91
PostgresEventStore.php:113
PostgresEventStore.php:132
PostgresEventStore.php:325
Projection/MySqlProjectionManager.php:124
Projection/MySqlProjectionManager.php:137
Projection/MySqlProjectionManager.php:150
Projection/MySqlProjectionManager.php:283
Projection/MySqlProjectionManager.php:304
Projection/MySqlProjectionManager.php:325
Projection/PdoEventStoreProjector.php:337
Projection/PdoEventStoreProjector.php:359

I stop naming them all for now ;-)

File: src/Projection/PdoEventStoreProjector.php (hunk 2 above)
Comment (dragosprotung) [inline src/Projection/PdoEventStoreProjector.php]: There are no error codes checked there. Shall I check for errors and throw custom exception ?

File: src/Projection/PdoEventStoreReadModelProjector.php (hunk 3 above)
Comment (dragosprotung) [inline src/Projection/PdoEventStoreReadModelProjector.php]: There are no error codes checked there. Shall I check for errors and throw custom exception ?

**Comment by dragosprotung** [comment]
How should the missing try-catch be handled ? Shall I check any error codes ? (basically anything other than `00000`).  If I throw any custom exceptions, then current behavior will change. Or just ignore the exception and rely on checking the fetch result ?

File: src/Projection/PdoEventStoreReadModelProjector.php (hunk 3 above)
Comment (prolic) [inline src/Projection/PdoEventStoreReadModelProjector.php]: Sure why not. Just take care that some errors are ok (duplicates).

**Comment by prolic** [comment]
Adding custom exception would be nice. It's not a big change, you just get notification when something broke which usually gets ignored, which is kind of bad I think.

File: tests/TestUtil.php (hunk 1 above)
Comment (dragosprotung) [inline tests/TestUtil.php]: Done

File: src/Projection/PdoEventStoreReadModelProjector.php (hunk 3 above)
Comment (dragosprotung) [inline src/Projection/PdoEventStoreReadModelProjector.php]: updated for consistency

File: src/Projection/PdoEventStoreProjector.php (hunk 2 above)
Comment (dragosprotung) [inline src/Projection/PdoEventStoreProjector.php]: updated for consistency

**Comment by dragosprotung** [comment]
Added the rest of try/catch blocks. Hope I got all of them :)

**Comment by dragosprotung** [comment]
@prolic when do you think you will have time to review the changes ?

**Comment by prolic** [comment]
very much thanks for your work! really appreciate it!

maybe you want to add the same to https://github.com/prooph/pdo-snapshot-store/ ?

**Comment by dragosprotung** [comment]
Thanks.

I will make a PR soon

**Pull request merged**